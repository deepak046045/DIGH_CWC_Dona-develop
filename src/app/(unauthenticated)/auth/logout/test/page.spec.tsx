import { render } from '@testing-library/react';
import React from 'react';
import { signOut, useSession } from 'next-auth/react';
import { useRouter } from 'next/navigation';
import Logout from '../page';
import '@testing-library/jest-dom';

jest.mock('next-auth/react');
jest.mock('next/navigation', () => ({
  useRouter: jest.fn(),
}));

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ test: 100 }),
  })
) as jest.Mock;

const pushMock = jest.fn();

describe('Logout Page', () => {
  it('should render when unauthenticated', () => {
    (useSession as jest.Mock).mockReturnValueOnce({
      data: {},
      status: 'unauthenticated',
    });
    const { container } = render(<Logout />);
    expect(container).toMatchSnapshot();
  });

  it('should render when authenticated', () => {
    (useSession as jest.Mock).mockReturnValueOnce({
      data: {},
      status: 'authenticated',
    });
    (signOut as jest.Mock).mockResolvedValueOnce(true);
    (useRouter as jest.Mock).mockReturnValue({
      query: {},
      push: pushMock,
    });
    const { container } = render(<Logout />);
    expect(container).toMatchSnapshot();
  });
});
