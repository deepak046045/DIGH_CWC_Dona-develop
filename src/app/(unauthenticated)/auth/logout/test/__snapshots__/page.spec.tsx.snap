// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout Page should render when authenticated 1`] = `
<div>
  <div
    class="logout-container"
  >
    <h1>
      Redirecting to logout...
    </h1>
    <lds-loading-spinner
      animationspeed="1000"
      arialabel="loading"
      size="40"
    />
  </div>
</div>
`;

exports[`Logout Page should render when unauthenticated 1`] = `
<div>
  <div
    class="lds-card"
  >
    <div
      class="content-container"
    >
      <h1>
        You have been signed out.
      </h1>
    </div>
    <div
      class="h2-container"
    >
      <h2>
        To continue using this site, you will need to sign in again.
      </h2>
    </div>
  </div>
</div>
`;
